HousingRepairsSchedulingApi.Tests/GatewaysTests/DrsAppointmentGatewayTests.cs

```diff
@@ -618,23 +620,56 @@ public async void GivenAnEndDateEarlierThanTheStartDate_WhenExecute_ThenInvalidE
 
         [Fact]
 #pragma warning disable CA1707
-        public async void GivenValidArguments_WhenExecute_ThenBookingReferenceIsReturned()
+        public async void GivenValidArguments_WhenExecute_ThenOrderIsReturned()
 #pragma warning restore CA1707
         {
             // Arrange
-            const int bookingId = 12345;
+            var bookingReference = "10003829";
+            var startDateTime = new DateTime(2022, 05, 01);
 
             drsServiceMock.Setup(x =>
-                x.CreateOrder(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>())
-            ).ReturnsAsync(bookingId);
+                x.SelectOrder(It.IsAny<int>(), It.IsAny<DateTime?>())
+            ).ReturnsAsync(new order { orderId = 863256, theBookings = new booking[] { new booking { bookingId = 12345 } } });
 
             // Act
-            var startDateTime = new DateTime(2022, 05, 01);
-            var actual = await systemUnderTest.BookAppointment(BookingReference, SorCode, LocationId,
+            var actual = await systemUnderTest.BookAppointment(bookingReference, SorCode, LocationId,
                 startDateTime, startDateTime.AddDays(1));
 
             // Assert
-            Assert.Equal(BookingReference, actual);
-        }
+            Assert.Equal(bookingReference, actual);
+        }
+
+        public static IEnumerable<object[]> InvalidOrderTestData()
+        {
+            yield return new object[] { new DrsException(), "10003829", new DateTime(2022, 05, 01), (order)null };
+            yield return new object[] { new DrsException(), "10003829", new DateTime(2022, 05, 01), new order { orderComments = "No bookings in this order" } };
+            yield return new object[] { new DrsException(), "10003829", new DateTime(2022, 05, 01), new order { theBookings = new booking[] { new booking { bookingId = 0 } } } };
+        }

```

Comment:
Never thought of this. I've always found it annoying that I couldn't create objects as test parameters.